src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -196,6 +239,20 @@ def test_ad_scalar_type_no_inc():  # pylint: disable=invalid-name
             "(['gh_sum']) but found 'gh_inc'" in str(excinfo.value))
 
 
+def test_ad_int_scalar_type_no_inc():  # pylint: disable=invalid-name
```

Comment:
I guess "previous" means "test_ad_real_scalar_type_no_inc". Merged using a loop to test for all scalars in VALID_SCALAR_NAMES.

Instead of extending "test_ad_real_scalar_type_no_inc" to test that we reject GH_SUM for integer scalars I created a new test "test_ad_int_scalar_type_no_sum()" (different metadata descriptors).

As noted above, this test effectively checks for the same thing as test_scalar_int_builtin_error(monkeypatch)